docs/sc_FunctionalRequirements.csv

```diff
@@ -8,114 +8,114 @@ d)       ATS Command Error Counter
+SC3005,SC3005,"Upon receipt of a Disable Continuation Execution of ATS On Error Command, SC shall set the Continuation Execution of ATS On Error Flag to DISABLED.","Provides control on how serious a data integrity check failure is – in this case, terminate the current ATS."
 SC4000,SC4000,"Upon receipt of a Start RTS Command, SC shall execute the command-specified RTS, or range of RTS, provided all of the following conditions are met:
 a) The command-specified RTS, or range of RTS, is not currently executing
 b) The RTS, or range of RTS, table(s) is Enabled 
-c) The RTS, or range of RTS, table(s) has been Loaded",Method for starting an RTS.
+c) The RTS, or range of RTS, table(s) has been Loaded",Method for starting an RTS.
 SC4000.1,SC4000.1,"If the conditions are met, SC shall issue an event message indicating the RTS started if the RTS number is less than <PLATFORM_DEFINED> RTS number.","Provides the ground an indication that an RTS started.
-Having a config parameter allows a range of RTSs to be specified such that no event message is sent."
-SC4000.2,SC4000.2,"If the conditions are not met, SC shall reject the command and send an event message.",Ground needs to know if an RTS was not executed.
-SC4001,SC4001,"SC shall dispatch commands within the RTS table, in position order, as the relative time-tag specified in the RTS command expires.",RTS contain relative times between commands.
-SC4001.1,SC4001.1,The time-tag shall be interpreted as the number of seconds to delay relative to the previous RTS command dispatched from that RTS table.,
-SC4001.2,SC4001.2,"For the first command in an RTS table, the delay time shall be relative to the receipt of the RTS Start Command.",Provides the ability to delay the executing of the first command.
+Having a config parameter allows a range of RTSs to be specified such that no event message is sent."
+SC4000.2,SC4000.2,"If the conditions are not met, SC shall reject the command and send an event message.",Ground needs to know if an RTS was not executed.
+SC4001,SC4001,"SC shall dispatch commands within the RTS table, in position order, as the relative wakeup count specified in the RTS command expires.",RTS contain relative wakeup counts between commands.
+SC4001.1,SC4001.1,The wakeup count shall be interpreted as the number of wakeup events to delay relative to the previous RTS command dispatched from that RTS table.,
+SC4001.2,SC4001.2,"For the first command in an RTS table, the delay shall be relative to the receipt of the RTS Start Command.",Provides the ability to delay the executing of the first command.
 SC4001.3,SC4001.3,"Prior to the dispatch of each individual RTS command, SC shall verify the validity of the following command parameters:

```

Comment:
I would add a new requirement that says "If an ATS and RTS both meet the criteria to execute, the ATS will be executed and the RTS will be delayed until the next wakeup."

That locks in the behavior we have now so it's not a surprise, and if we ever want to change that behavior we know the baseline.

Reply:
@RichLandau is that true?  Do you mean the command to start the ATS/RTS, or if commands within the ATS/RTS are would go out the same wakeup?  I think in SC_OneHzWakeupCmd (at least the version before this change) it'll process up to `SC_MAX_CMDS_PER_SEC` from either ATS or RTS.

Reply:
Good point, I was looking at SC_UpdateNextTime in the diff without seeing the loop it was in. Looking here:
https://github.com/nasa/SC/pull/146/files#diff-dacb535e36ef925e9071b7ca9d813f065b6eeeb756dbd506a2dc3743b004cd03R135

It looks like ATPs are processed first, and then (previously) RTPs were scheduled if they came before the ATP. The logic now schedules the RTP if there isn't an ATP scheduled. Do you think this will function the same as before?

Reply:
@RichLandau - I don't think so, although the logic confuses me a bit and seems unnecessarily complex.  Why not in the wakeup just do the switch if pending, send the Atp cmds in a loop until done or SC_MAX_CMDS reached, then send Rtp commands in a loop until done or SC_MAX_CMDS reached?

Reply:
I guess the Switch needs to be inside the Atp cmd loop since you can switch from an Atp... but otherwise, why not split into 2 loops?

Reply:
Apologies for the spamming... but maybe it does need to be in one loop if you can command an ATS switch from an RTS.  Even if that's the case the loop could still be simplified to switch, then send an absolute command if the next cmd time is  <= current time, and if not send an rts command if next wakeup is <= current wakeup.  Exit once either no command is sent or SC_MAX_CMDS is met.

Reply:
I don't completely understand your example, but I think what you are describing would be changing current behavior.  Right now if ATS and RTS are both due in a wakeup, it does ATS first then cycles through all the RTS's in order until all due commands are sent or it hits the command limit.  ATS command is higher priority than RTS 5 is higher priority than RTS 20.  There isn't a calculation at a lower level of if the ATS absolute time is actually before or after the RTS... just that they are due on the same wakeup.